### DISCUSSION TITLE
[ZK Stack] - Can't start zk stack in the end of the wizard

### DISCUSSION THREAD

**Discussion by nonameb3** [zkSync-Community-Hub/zksync-developers#260] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

rustc 1.74.0-nightly

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

When I initial zk stack by useing `zk stack init`, I expect running zk with new chain config bug I got error in the end of the wizard. 

### Expected Behavior

CLI start a new zk server that I can use it RPC.

### Current Behavior

Got error message in the end of the wizard in the step start server. Whatever I choose Yes or Not right now and run 
`zk server --components "http_api,tree_lightweight,eth,data_fetcher,state_keeper,housekeeper"` it not working.

```
> Start server
? Do you want to start your hyperchain server now? … 
❯ Yes (default components)
  Yes (custom components)
  Not right now (you can now configure prover, generate docker files, or just run the server later)
```

### Repository Link (if applicable)

https://github.com/matter-labs/zksync-era

### Additional Details

- Use latest code from main branch.

Log of the wizard
`> Start server
✔ Do you want to start your hyperchain server now? · Yes (default components)
    Finished release [optimized] target(s) in 2.19s
    
     Running `target/release/zksync_server --components=http_api,eth,state_keeper,housekeeper,tree`
2024-01-05T11:04:32.673711Z  INFO zksync_server: No sentry URL was provided
thread 'main' panicked at /Users/roonnapai/Documents/tokenx/zkStack/zksync-era/core/lib/zksync_core/src/lib.rs:181:10:
failed to build connection_pool: Failed connecting to database

Caused by:
    0: error communicating with database: failed to lookup address information: nodename nor servname provided, or not known
    1: failed to lookup address information: nodename nor servname provided, or not known

Stack backtrace:
   0: std::backtrace_rs::backtrace::libunwind::trace
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1: std::backtrace_rs::backtrace::trace_unsynchronized
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2: std::backtrace::Backtrace::create
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/backtrace.rs:331:13
   3: <E as anyhow::context::ext::StdError>::ext_context
   4: zksync_dal::connection::ConnectionPoolBuilder::build::{{closure}}
   5: zksync_server::main::{{closure}}
   6: tokio::runtime::park::CachedParkThread::block_on
   7: tokio::runtime::context::runtime::enter_runtime
   8: tokio::runtime::runtime::Runtime::block_on
   9: zksync_server::main
  10: std::sys_common::backtrace::__rust_begin_short_backtrace
  11: std::rt::lang_start::{{closure}}
  12: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/ops/function.rs:284:13
  13: std::panicking::try::do_call
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:524:40
  14: std::panicking::try
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:488:19
  15: std::panic::catch_unwind
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panic.rs:142:14
  16: std::rt::lang_start_internal::{{closure}}
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/rt.rs:148:48
  17: std::panicking::try::do_call
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:524:40
  18: std::panicking::try
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:488:19
  19: std::panic::catch_unwind
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panic.rs:142:14
  20: std::rt::lang_start_internal
             at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/rt.rs:148:20
  21: _main
stack backtrace:
   0:        0x103e18cf0 - std::backtrace_rs::backtrace::libunwind::trace::h6349c5baaf0e6736
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:        0x103e18cf0 - std::backtrace_rs::backtrace::trace_unsynchronized::h0777b695467c5969
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:        0x103e18cf0 - std::sys_common::backtrace::_print_fmt::h37a26013c7e2fe6f
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:67:5
   3:        0x103e18cf0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hfc45a9cfd936fd12
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:44:22
   4:        0x103e39da4 - core::fmt::rt::Argument::fmt::h118095ac315d673a
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/fmt/rt.rs:138:9
   5:        0x103e39da4 - core::fmt::write::h8a2f9b844b656b2a
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/fmt/mod.rs:1094:21
   6:        0x103e1575c - std::io::Write::write_fmt::h48feeb430e5b9178
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/io/mod.rs:1714:15
   7:        0x103e18b30 - std::sys_common::backtrace::_print::h2732436374d5ca92
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:47:5
   8:        0x103e18b30 - std::sys_common::backtrace::print::hb462cd3ae7cd2cac
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:34:9
   9:        0x103e1a508 - std::panicking::panic_hook_with_disk_dump::{{closure}}::h2aefaaff1382e42d
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:278:22
  10:        0x103e1a208 - std::panicking::panic_hook_with_disk_dump::hc7dbf841ebac63a7
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:312:9
  11:        0x103e1aab8 - std::panicking::default_hook::hb3fbd6d64db320d9
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:239:5
  12:        0x103e1aab8 - std::panicking::rust_panic_with_hook::h7a646a0d2cc797f8
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:729:13
  13:        0x103e1a9c0 - std::panicking::begin_panic_handler::{{closure}}::h0e1c0a22168ab174
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:621:13
  14:        0x103e1917c - std::sys_common::backtrace::__rust_end_short_backtrace::h47b6e82934f326fa
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:170:18
  15:        0x103e1a724 - rust_begin_unwind
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:617:5
  16:        0x103f104ec - core::panicking::panic_fmt::h7583d88e5fa9ffcd
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/panicking.rs:67:14
  17:        0x103f10814 - core::result::unwrap_failed::h78a2af4fb8daedfb
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/result.rs:1652:5
  18:        0x1026a60d8 - zksync_server::main::{{closure}}::h7fc32f23d9a838a3
  19:        0x10269a9e8 - tokio::runtime::park::CachedParkThread::block_on::h7fa3429646d3b2f5
  20:        0x1025eab60 - tokio::runtime::context::runtime::enter_runtime::h5b738edd5f5ad6df
  21:        0x1027a3fd4 - tokio::runtime::runtime::Runtime::block_on::h9e5fa992e3532417
  22:        0x102537174 - zksync_server::main::h76304dc7614c199d
  23:        0x1027e4cd4 - std::sys_common::backtrace::__rust_begin_short_backtrace::hf563a94f7d753898
  24:        0x10282cdf8 - std::rt::lang_start::{{closure}}::hbaf063e4751a51e5
  25:        0x103e0eb40 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h30b9c905d4d7a3f1
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/ops/function.rs:284:13
  26:        0x103e0eb40 - std::panicking::try::do_call::hd229b40a2ffa873a
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:524:40
  27:        0x103e0eb40 - std::panicking::try::h520f5809142519aa
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:488:19
  28:        0x103e0eb40 - std::panic::catch_unwind::h1a8cb6f017912545
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panic.rs:142:14
  29:        0x103e0eb40 - std::rt::lang_start_internal::{{closure}}::h814bb4873d109c7c
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/rt.rs:148:48
  30:        0x103e0eb40 - std::panicking::try::do_call::h8e1c1746cdde7a6e
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:524:40
  31:        0x103e0eb40 - std::panicking::try::hf1a2299f115c7b8f
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:488:19
  32:        0x103e0eb40 - std::panic::catch_unwind::h257fdc2f43b2abf3
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panic.rs:142:14
  33:        0x103e0eb40 - std::rt::lang_start_internal::h36385bef630d9072
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/rt.rs:148:20
  34:        0x10253727c - _main
Error: Child process exited with code 101`

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by zk-Lumi** [comment, marked as answer]
Thanks for reporting @nonameb3 :)

**Temporary Solution ->**

Change `hyperchain_wizard.ts` + `hyperchain_wizard.js` to use localhost instead of geth + postgres urls ->

`http://geth:8545` 
-> 
`http://localhost:8545`

`postgres://postgres:notsecurepassword@postgres:5432/zksync_local` 
-> 
`postgres://postgres:notsecurepassword@localhost:5432/zksync_local`

<img width="864" alt="Screenshot" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/149794418/b8ab033d-8571-4158-b43d-1a6c3c6c1714">

Then re run zk stack init to create new chan with env file regenerated

Let me know if if that works for you.


We are further investigating this issue and will keep this thread updated.

**Reply by nonameb3** [reply]
Thanks, It works!

**Comment by ternianitha** [comment]
It seems like the error is related to the database connection issue during the initialization of the zk stack. The error message indicates that the server failed to connect to the database due to a lookup address failure. This could be caused by various reasons such as incorrect database configuration or network issues. To resolve this issue, you should check your database configuration and ensure that the database server is running and accessible from the zk server. Additionally, verify the network settings to ensure that there are no connectivity issues. If the issue persists, you may need to seek further assistance from the development team or community forums for troubleshooting steps specific to the zk stack.

**Comment by franksofa** [comment]
The error message indicates a failure to connect to the database, resulting in the zk server startup process terminating unexpectedly. This issue might stem from misconfiguration or a problem with the database setup. Ensure that the database configuration is correctly specified and that the database service is running and accessible. Additionally, check for any firewall rules or network restrictions that might prevent the server from connecting to the database. If the issue persists, consider reviewing the database connection code in the zk server source files for any potential bugs or errors.

**Comment by stennettmeisha** [comment]
Error encountered while starting the zkSync server after initializing the zk stack with Sepolia as the base layer. The error indicates a failure to connect to the database, resulting in a panic. Further investigation and troubleshooting are needed to resolve this issue.

**Comment by propmacboorsp** [comment]
It seems that the error is related to the zkSync server failing to connect to the database. The error message indicates a problem with address resolution, suggesting a potential issue with the database configuration or network settings. You may need to double-check your database configuration and ensure that the server can resolve the hostname specified for the database connection. Additionally, verify that the database service is running and accessible from the server where zkSync is deployed.
